ui/src/importer/importerTaskDetails/ImportedDocuments.js

```diff
@@ -97,163 +245,92 @@ export class ImportedDocuments extends React.Component {
-                />
-              </Table.HeaderCell>
-            </Table.Row>
-          </Table.Footer>
-        </Table>
-        <JsonViewModal />
-        <SeriesImportDetailsModal />
-        <EitemImportDetailsModal />
-      </>
+      <Statistic.Group widths="seven">
+        {Object.keys(statistics).map(function(statistic, index) {
+          return (
+            <Statistic
+              className={
+                selectedResult === statistic
+                  ? 'import-statistic statistic-selected'
+                  : 'import-statistic'
+              }
+              key={statistic}
+              onClick={() => this.currentFilterResults(statistic)}
+            >
+              <Statistic.Value>{statistics[statistic].value}</Statistic.Value>
+              <Statistic.Label>{statistics[statistic].text}</Statistic.Label>
+            </Statistic>
+          );
+        }, this)}
+      </Statistic.Group>
     );
   };
 
   renderCancelButton = () => {
     const { data } = this.state;
     const { taskId } = this.props;
+    const isRunning = !_isEmpty(data) && data.status === 'RUNNING';
 
-    return (
+    return isRunning ? (
       <>
-        {!_isEmpty(data) && data.status === 'RUNNING' && (
-          <CancelImportTask logId={taskId} />
-        )}
+        <CancelImportTask logId={taskId} /> <Icon loading name="circle notch" />
+        This may take a while. You may leave the page, the process will continue
+        in background.
       </>
-    );
+    ) : null;
   };
 
   render() {
-    const { data } = this.state;
+    const {
+      data,
+      mode,
+      importCompleted,
+      status,
+      currentFilteredRecords,
+    } = this.state;
     return (
       <>
         {this.renderImportReportHeader()}
         {!_isEmpty(data) && data.status !== 'FAILED' ? (
           <>
-            <Header as="h3">Import report</Header>
-            {modeFormatter(data.mode)}{' '}
-            {!_isEmpty(data) ? (
-              (data.loaded_entries || data.loaded_entries === 0) &&
-              data.entries_count ? (
-                <span>
-                  {'Processed ' +
-                    data.loaded_entries +
-                    ' records out of ' +
-                    data.entries_count +
-                    '.'}
-                </span>
-              ) : (
-                <span>Processing file...</span>
+            <Segment>
+              {!_isEmpty(data) ? (
+                (data.loaded_entries || data.loaded_entries === 0) &&
+                data.entries_count ? (
+                  <Grid className="middle aligned">
+                    <Grid.Column width={3}>
+                      <Segment>
+                        Mode: {mode}
+                        <Divider />
+                        Status: {status}
+                      </Segment>
+                    </Grid.Column>
+                    <Grid.Column width={13}>
+                      {this.renderStatistics()}
```

Comment:
does it make sense to have a separate component here instead of method? I would like to avoid the JSX code spread around in the methods different than render if possible